src/test/java/net/starlark/java/eval/testdata/dict.star

```diff
@@ -43,6 +43,10 @@ dict().popitem() ### dictionary is empty
 ---
 dict(a=2).pop('z') ### KeyError: "z"
 ---
+d = {}
+unsafe_shallow_freeze(d)
+d.pop("nonexistent", "default") ### trying to mutate a frozen dict value
```

Comment:
Do we have test to ensure that `{}.pop([], 1)` fails due to unhashable key, rather than returning 1 because the dict is empty and thus trivially does not contain the key?

The spec is indeed a little unclear, but I agree with your interpretation.